core/src/main/java/org/apache/accumulo/core/fate/zookeeper/ZooReader.java

```diff
@@ -184,7 +184,11 @@ protected <R> R retryLoopMutator(ZKFunctionMutator<R> zkf,
       try {
         return zkf.apply(getZooKeeper());
       } catch (KeeperException e) {
-        if (alwaysRetryCondition.test(e) || useRetryForTransient(retries, e)) {
+        if (alwaysRetryCondition.test(e)) {
+          retries.waitForNextAttempt(log,
```

Comment:
Looking at RETRY_FACTORY at the top of the class it constructs a retry with `maxRetries(10)`.  Would this mean a different retry factory is needed for the always retry case?

Reply:
The maxRetries sets the counter. Calling the waitForNextAttempt does not decrement the counter. You have to call. `retries.useRetry()` to decrement the counter. This patch waits for the next retry using the backoff, but does not call `retries.useRetry()`.